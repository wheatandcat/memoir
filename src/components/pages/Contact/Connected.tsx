import React, { memo, useCallback, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import TemplateContact from 'components/templates/Contact/Page';
import { useRecoilValue } from 'recoil';
import { userState } from 'store/atoms';
import * as Device from 'expo-device';
import { Platform, Alert, TouchableOpacity } from 'react-native';
import Constants from 'expo-constants';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';

type Props = {};

const url = Constants.expoConfig?.extra?.INQUIRY_API || '';

export type ConnectedType = {
  userID: string;
  loading: boolean;
  send: boolean;
  text: string;
  onContact: (text: string) => void;
  onChangeText: (text: string) => void;
  onClose: () => void;
};

type Request = {
  body: string;
  name: string;
  email: string;
  env: string;
  userID: string;
  device: string;
  category: string;
};

const Connected: React.FC<Props> = () => {
  const user = useRecoilValue(userState);
  const navigation = useNavigation();
  const [loading, setLoading] = useState(false);
  const [send, setSend] = useState(false);
  const [text, setText] = useState('');

  const onContact = useCallback(async () => {
    if (text.length === 0) {
      Alert.alert('コメントが入力されていません');
      return;
    }

    if (loading) {
      return;
    }

    setLoading(true);

    const req: Request = {
      body: text,
      name: user.displayName || '無名',
      email: 'foobazbar@gmail.com',
      userID: user.userID || '無し',
      env: 'アプリ',
      device: `${Platform.OS}/${Platform.Version}/${Device.modelName}/${String(
        Device.osInternalBuildId
      )}`,
      category: 'フィードバックを送信',
    };

    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(req),
    });

    if (!response.ok) {
      setLoading(false);
      Alert.alert('送信に失敗しました');
      return;
    }

    setSend(true);
  }, [user, text, loading]);

  const onClose = useCallback(() => {
    setSend(false);
    navigation.goBack();
  }, [navigation]);

  React.useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: () => (
        <View pr={2} mr={1}>
          <TouchableOpacity onPress={onContact}>
            <Text>送信</Text>
          </TouchableOpacity>
        </View>
      ),
    });
  }, [navigation, onContact]);

  return (
    <TemplateContact
      send={send}
      userID={user.id ?? ''}
      loading={loading}
      onContact={onContact}
      text={text}
      onChangeText={setText}
      onClose={onClose}
    />
  );
};

export default memo(Connected);
