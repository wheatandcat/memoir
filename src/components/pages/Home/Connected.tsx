import React, { memo, useCallback, useState } from 'react';
import { useRecoilState, useRecoilValue } from 'recoil';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { useCreateItemMutation, NewItem } from 'queries/api/index';
import { homeDateState, homeItemsState, Item } from 'store/atoms';
import useHomeItems from 'hooks/useHomeItems';
import { Props as IndexProps } from './';
import Plain from './Plain';

dayjs.locale('ja');
dayjs.extend(advancedFormat);

type Props = IndexProps & {
  openSettingModal: boolean;
  onCloseSettingModal: () => void;
};

type State = {
  openAddItemModal: boolean;
};

export type ConnectedType = {
  items: Item[];
  addItemLoading: boolean;
  date: string;
  openAddItemModal: boolean;
  openSettingModal: boolean;
  onAddItem: (item: NewItem) => void;
  onChangeDate: (date: string) => void;
  onCloseAddItem: () => void;
  onCloseSettingModal: () => void;
  onItem: (itemID: string) => void;
  onMemoir: () => void;
  onOpenAddItem: () => void;
};

const initialState = (): State => ({
  openAddItemModal: false,
});

const Connected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());
  const [homeDate, setHomeDate] = useRecoilState(homeDateState);
  const homeItems = useRecoilValue(homeItemsState);
  const { loading, error, refetch } = useHomeItems();

  const onOpenAddItem = useCallback(() => {
    setState((s) => ({ ...s, openAddItemModal: true }));
  }, []);

  const onCloseAddItem = useCallback(() => {
    setState((s) => ({ ...s, openAddItemModal: false }));
  }, []);

  const [createItemMutation, createItemMutationData] = useCreateItemMutation({
    async onCompleted() {
      await refetch?.();

      onCloseAddItem();
    },
  });

  const onAddItem = useCallback(
    (newItem: NewItem) => {
      const variables = {
        input: newItem,
      };

      createItemMutation({ variables });
    },
    [createItemMutation]
  );

  const onChangeDate = useCallback(
    (date: string) => {
      setHomeDate({
        date: dayjs(date).format('YYYY-MM-DDT00:00:00+09:00'),
      });
    },
    [setHomeDate]
  );

  const onItem = useCallback(
    (itemID: string) => {
      props.navigation.navigate('ItemDetail', {
        id: itemID,
        date: homeDate.date,
      });
    },
    [props.navigation, homeDate.date]
  );

  const onMemoir = useCallback(() => {
    props.navigation.navigate('Memoir', {
      startDate: dayjs().add(-7, 'day').format('YYYY-MM-DDT00:00:00+09:00'),
      endDate: dayjs().format('YYYY-MM-DDT00:00:00+09:00'),
    });
  }, [props.navigation]);

  return (
    <Plain
      items={homeItems.items}
      loading={loading}
      error={error}
      addItemLoading={createItemMutationData.loading}
      date={dayjs(homeDate.date).format('YYYY-MM-DD')}
      openAddItemModal={state.openAddItemModal}
      openSettingModal={props.openSettingModal}
      onAddItem={onAddItem}
      onChangeDate={onChangeDate}
      onCloseAddItem={onCloseAddItem}
      onCloseSettingModal={props.onCloseSettingModal}
      onItem={onItem}
      onMemoir={onMemoir}
      onOpenAddItem={onOpenAddItem}
    />
  );
};

export default memo(Connected);
