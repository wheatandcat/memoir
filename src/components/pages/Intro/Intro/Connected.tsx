import React, { memo, useCallback, useState } from 'react';
import * as Notifications from 'expo-notifications';
import useMemoirNotificationSetting from 'hooks/useMemoirNotificationSetting';
import { useNotification } from 'containers/Notification';
import Plain from './Plain';

export type Props = {
  onFinish: () => void;
};

export type ConnectedType = {
  step: number;
  onSaveNotification: (input: Input, callback: () => void) => void;
  onStep: (num: number) => void;
};

type Input = {
  dayOfWeek: number;
  hours: number;
  minutes: number;
  notification: boolean;
};

const Connected: React.FC<Props> = () => {
  const memoirNotificationSetting = useMemoirNotificationSetting();
  const { onPermissionRequest } = useNotification();

  const [step, setStep] = useState(0);

  const onStep = useCallback((num: number) => {
    setStep(num);
  }, []);

  const onSaveNotification = useCallback(
    async (input: Input, callback: () => void) => {
      if (input.notification) {
        if (!onPermissionRequest) {
          return;
        }

        onPermissionRequest(() => {
          Notifications.scheduleNotificationAsync({
            content: {
              body: 'ふりかえりの時間になりました',
              data: {
                urlScheme: 'Memoir',
              },
            },
            trigger: {
              hour: input.hours,
              minute: input.minutes,
              weekday: input.dayOfWeek,
              repeats: true,
            },
          });
        });

        // NOTE: API出力
      } else {
        // 通知OFFの場合は、通知をキャンセルする
        Notifications.cancelAllScheduledNotificationsAsync();
      }

      memoirNotificationSetting.onSave(input);
      callback();
    },
    [memoirNotificationSetting, onPermissionRequest]
  );

  return (
    <Plain
      loading={memoirNotificationSetting.loading}
      data={{
        dayOfWeek: memoirNotificationSetting.dayOfWeek,
        hours: memoirNotificationSetting.hours,
        minutes: memoirNotificationSetting.minutes,
        notification: memoirNotificationSetting.notification,
      }}
      onSaveNotification={onSaveNotification}
      step={step}
      onStep={onStep}
    />
  );
};

export default memo(Connected);
