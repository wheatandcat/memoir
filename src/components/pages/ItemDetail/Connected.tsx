import React, { memo, useCallback, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import { useSetRecoilState } from 'recoil';
import dayjs from 'lib/dayjs';
import {
  ItemDocument,
  UpdateItemDocument,
  DeleteItemDocument,
  UpdateItem,
  NewItem,
  DeleteItem,
} from 'queries/api/index';
import { homeDateState } from 'store/atoms';
import useHomeItems from 'hooks/useHomeItems';
import { useQuery, useMutation } from '@apollo/client';
import Plain from './Plain';

export type Props = {
  itemID: string;
  date: string;
};

type State = {
  openUpdateItemModal: boolean;
};

export type ConnectedType = {
  updateItemLoading: boolean;
  date: string;
  openUpdateItemModal: boolean;
  onOpenUpdateItem: () => void;
  onChangeDate: (date: string) => void;
  onUpdateItem: (updateItem: NewItem) => void;
  onDeleteItem: () => void;
  onCloseUpdateItem: () => void;
};

const initialState = (): State => ({
  openUpdateItemModal: false,
});

const Connected: React.FC<Props> = (props) => {
  const navigation = useNavigation();
  const [state, setState] = useState<State>(initialState());
  const setHomeDate = useSetRecoilState(homeDateState);
  const homeItems = useHomeItems();

  const { loading, data, error, refetch } = useQuery(ItemDocument, {
    variables: {
      id: props.itemID,
    },
  });

  const onOpenUpdateItem = useCallback(() => {
    setState((s) => ({ ...s, openUpdateItemModal: true }));
  }, []);

  const onCloseUpdateItem = useCallback(() => {
    setState((s) => ({ ...s, openUpdateItemModal: false }));
  }, []);

  const [updateItemMutation] = useMutation(UpdateItemDocument, {
    async onCompleted() {
      await refetch?.();

      onCloseUpdateItem();
    },
  });

  const onUpdateItem = useCallback(
    (newItem: NewItem) => {
      const updateItem: UpdateItem = {
        id: props.itemID,
        ...newItem,
      };

      const variables = {
        input: updateItem,
      };

      updateItemMutation({ variables });
    },
    [updateItemMutation, props.itemID]
  );

  const [deleteItemMutation] = useMutation(DeleteItemDocument, {
    async onCompleted() {
      await homeItems?.refetch?.();
      navigation.goBack();
    },
  });

  const onDeleteItem = useCallback(() => {
    const deleteItem: DeleteItem = {
      id: props.itemID,
    };
    const variables = {
      input: deleteItem,
    };

    deleteItemMutation({ variables });
  }, [props.itemID, deleteItemMutation]);

  const onChangeDate = useCallback(
    (date: string) => {
      const formatDate = dayjs(props.date).format('YYYY-MM-DD');
      if (formatDate !== date) {
        setHomeDate({
          date: dayjs(date).format('YYYY-MM-DDT00:00:00+09:00'),
        });

        navigation.goBack();
      }
    },
    [props, navigation, setHomeDate]
  );

  return (
    <Plain
      data={data}
      loading={loading}
      error={error}
      updateItemLoading={false}
      date={dayjs(props.date).format('YYYY-MM-DD')}
      openUpdateItemModal={state.openUpdateItemModal}
      onChangeDate={onChangeDate}
      onOpenUpdateItem={onOpenUpdateItem}
      onUpdateItem={onUpdateItem}
      onDeleteItem={onDeleteItem}
      onCloseUpdateItem={onCloseUpdateItem}
    />
  );
};

export default memo(Connected);
