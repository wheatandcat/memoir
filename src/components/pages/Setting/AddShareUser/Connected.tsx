import React, { memo, useCallback } from 'react';
import { Alert } from 'react-native';
import {
  InviteQuery,
  InviteByCodeQueryVariables,
  CreateRelationshipRequestMutationVariables,
  CreateRelationshipRequestMutation,
  InviteDocument,
  InviteByCodeDocument,
  UpdateInviteDocument,
  CreateInviteDocument,
  CreateRelationshipRequestDocument,
} from 'queries/api/index';
import { useRecoilValue } from 'recoil';
import { userState, User } from 'store/atoms';
import { useQuery, useLazyQuery, useMutation } from '@apollo/client';
import Plain from './Plain';

export type Props = {};

export type Invite = InviteQuery['invite'];

export type ConnectedType = {
  user: User;
  requestUser:
    | CreateRelationshipRequestMutation['createRelationshipRequest']['user']
    | null;
  onCreateInvite: () => void;
  onUpdateInvite: () => void;
  onSearchInviteCode: (code: string) => void;
  onCreateRelationshipRequest: (code: string) => void;
  creating: boolean;
  updating: boolean;
  loading: boolean;
  requesting: boolean;
  confirmUser: {
    id: string;
    displayName: string;
    image: string;
  } | null;
};

const Connected: React.FC<Props> = () => {
  const { loading, data, error, refetch } = useQuery(InviteDocument);
  const [getInviteByCode, inviteByCodeData] = useLazyQuery(
    InviteByCodeDocument,
    {
      fetchPolicy: 'network-only',
    }
  );

  const [createRelationshipRequestMutation, relationshipRequestMutationData] =
    useMutation(CreateRelationshipRequestDocument);

  const user = useRecoilValue(userState);
  const [createInviteMutation, createInviteMutationData] = useMutation(
    CreateInviteDocument,
    {
      onCompleted() {
        refetch();
      },
      onError() {
        Alert.alert('エラー', '作成に失敗しました');
      },
    }
  );
  const [updateInviteMutation, updateInviteMutationData] = useMutation(
    UpdateInviteDocument,
    {
      onCompleted() {
        refetch();
      },
      onError() {
        Alert.alert('エラー', '更新に失敗しました');
      },
    }
  );

  const onCreateInvite = useCallback(() => {
    createInviteMutation();
  }, [createInviteMutation]);

  const onUpdateInvite = useCallback(() => {
    updateInviteMutation();
  }, [updateInviteMutation]);

  const onSearchInviteCode = useCallback(
    (code: string) => {
      const variables: CreateRelationshipRequestMutationVariables = {
        input: {
          code,
        },
      };
      createRelationshipRequestMutation({ variables });
    },
    [createRelationshipRequestMutation]
  );

  const onCreateRelationshipRequest = useCallback(
    (code: string) => {
      const variables: InviteByCodeQueryVariables = {
        code,
      };
      getInviteByCode({ variables });
    },
    [getInviteByCode]
  );

  return (
    <Plain
      loading={loading}
      error={error}
      data={data}
      user={user}
      onCreateInvite={onCreateInvite}
      onUpdateInvite={onUpdateInvite}
      onSearchInviteCode={onSearchInviteCode}
      onCreateRelationshipRequest={onCreateRelationshipRequest}
      confirmUser={inviteByCodeData.data?.inviteByCode ?? null}
      creating={createInviteMutationData.loading}
      updating={updateInviteMutationData.loading}
      requesting={relationshipRequestMutationData.loading}
      requestUser={
        relationshipRequestMutationData.data?.createRelationshipRequest?.user ??
        null
      }
    />
  );
};

export default memo(Connected);
