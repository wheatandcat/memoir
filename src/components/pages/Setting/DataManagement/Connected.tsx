import React, { memo, useCallback } from 'react';
import {
  useDeleteUserMutation,
  useRelationshipsQuery,
} from 'queries/api/index';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import useFirebaseAuth from 'hooks/useFirebaseAuth';
import Loading from 'components/atoms/Loading';
import { deleteImageAsync } from 'lib/image';
import { userState, screenState } from 'store/atoms';
import Plain from './Plain';

export type Props = {};

export type ConnectedType = {
  loading: boolean;
  disabledDeleteButton: boolean;
  onDelete: () => void;
};

const Connected: React.FC<Props> = () => {
  const user = useRecoilValue(userState);
  const setScreenState = useSetRecoilState(screenState);
  const { setupAuth, onLogout } = useFirebaseAuth(true);
  const relationshipsQuery = useRelationshipsQuery({
    variables: {
      input: {
        after: '',
        first: 1,
      },
      skip: true,
    },
  });
  const [deleteUserMutation, deleteUserMutationData] = useDeleteUserMutation({
    async onCompleted() {
      if (user.image) {
        await deleteImageAsync(user.image);
      }
      await onLogout();
      setScreenState({ seeYouAgain: true });
    },
  });

  const onDelete = useCallback(() => {
    deleteUserMutation();
  }, [deleteUserMutation]);

  if (!setupAuth || relationshipsQuery.loading) {
    return <Loading />;
  }

  return (
    <Plain
      loading={deleteUserMutationData.loading}
      error={deleteUserMutationData.error}
      onDelete={onDelete}
      disabledDeleteButton={
        (relationshipsQuery?.data?.relationships?.edges ?? []).length > 0
      }
    />
  );
};

export default memo(Connected);
