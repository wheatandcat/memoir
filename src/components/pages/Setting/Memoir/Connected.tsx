import React, { memo, useCallback } from 'react';
import * as Notifications from 'expo-notifications';
import { useNavigation } from '@react-navigation/native';
import useMemoirNotificationSetting from 'hooks/useMemoirNotificationSetting';
import { useNotification } from 'containers/Notification';
import Plain from './Plain';

export type Props = {};

export type ConnectedType = {
  onSave: (input: Input) => void;
};

type Input = {
  dayOfWeek: number;
  hours: number;
  minutes: number;
  notification: boolean;
};

const Connected: React.FC<Props> = () => {
  const navigation = useNavigation();
  const memoirNotificationSetting = useMemoirNotificationSetting();
  const { onPermissionRequest } = useNotification();

  const onSave = useCallback(
    async (input: Input) => {
      if (input.notification) {
        if (!onPermissionRequest) {
          return;
        }

        const ok = await onPermissionRequest(() => null);
        if (ok) {
          await Notifications.cancelAllScheduledNotificationsAsync();

          const trigger = {
            channelId: 'memoir',
            hour: input.hours,
            minute: input.minutes,
            weekday: input.dayOfWeek,
            repeats: true,
          };

          try {
            await Notifications.scheduleNotificationAsync({
              content: {
                body: 'ふりかえりの時間になりました',
                data: {
                  urlScheme: 'Memoir',
                },
              },
              trigger,
            });
          } catch (e) {
            console.log('err:', e);
          }
        }
      } else {
        // 通知OFFの場合は、通知をキャンセルする
        await Notifications.cancelAllScheduledNotificationsAsync();
      }

      memoirNotificationSetting.onSave(input);

      navigation.goBack();
    },
    [navigation, memoirNotificationSetting, onPermissionRequest]
  );

  return (
    <Plain
      loading={memoirNotificationSetting.loading}
      data={{
        dayOfWeek: memoirNotificationSetting.dayOfWeek,
        hours: memoirNotificationSetting.hours,
        minutes: memoirNotificationSetting.minutes,
        notification: memoirNotificationSetting.notification,
      }}
      onSave={onSave}
    />
  );
};

export default memo(Connected);
