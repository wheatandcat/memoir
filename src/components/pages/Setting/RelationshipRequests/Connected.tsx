import React, { memo, useState, useCallback } from 'react';
import { useNavigation } from '@react-navigation/native';
import {
  RelationshipRequestsDocument,
  AcceptRelationshipRequestDocument,
  NgRelationshipRequestDocument,
  RelationshipRequestsQueryVariables as Variables,
} from 'queries/api/index';
import { v4 as uuidv4 } from 'uuid';
import useRelationshipRequestsPaging from 'hooks/useRelationshipRequestsPaging';
import { useQuery, useMutation } from '@apollo/client';
import Plain from './Plain';
import { ScreenNavigationProp as RelationshipRequestsScreenNavigationProp } from './';

export type Props = {
  onCallback: () => void;
};

export type ConnectedType = {
  acceptRequesting: boolean;
  ngRequesting: boolean;
  onLoadMore: (after: string | null) => void;
  onOK: (followedId: string) => void;
  onNG: (followedId: string) => void;
};

const Connected: React.FC<Props> = (props) => {
  const [endCursor, setEndCursor] = useState('');
  const [reloadKey, setReloadKey] = useState('');
  const navigation = useNavigation<RelationshipRequestsScreenNavigationProp>();

  const queryResult = useQuery(RelationshipRequestsDocument, {
    variables: {
      input: {
        first: 5,
        after: endCursor,
      },
      skip: false,
      reloadKey,
    } as Variables & { reloadKey: string },
    nextFetchPolicy: 'network-only',
  });

  const { items, pageInfo, reset } = useRelationshipRequestsPaging(
    queryResult,
    {
      merge: true,
    }
  );

  const [
    acceptRelationshipRequestMutation,
    acceptRelationshipRequestMutationData,
  ] = useMutation(AcceptRelationshipRequestDocument, {
    onCompleted(data) {
      reset();
      setEndCursor('');
      setReloadKey(uuidv4());
      props.onCallback();

      const followerId = data.acceptRelationshipRequest.followerId;
      const user = items.find((v) => v.followerId === followerId);

      navigation.navigate('SettingAcceptedRelationship', {
        displayName: user?.user?.displayName || '',
        image: user?.user?.image || '',
      });
    },
  });
  const [ngRelationshipRequestMutation, ngRelationshipRequestMutationData] =
    useMutation(NgRelationshipRequestDocument, {
      onCompleted() {
        reset();
        setEndCursor('');
        setReloadKey(uuidv4());
        props.onCallback();
      },
    });

  const onLoadMore = useCallback((after: string | null) => {
    setEndCursor(after || '');
  }, []);

  const onOK = useCallback(
    (followedId: string) => {
      if (
        acceptRelationshipRequestMutationData.loading ||
        ngRelationshipRequestMutationData.loading
      ) {
        return;
      }

      acceptRelationshipRequestMutation({
        variables: {
          followedID: followedId,
        },
      });
    },
    [
      acceptRelationshipRequestMutation,
      acceptRelationshipRequestMutationData.loading,
      ngRelationshipRequestMutationData.loading,
    ]
  );
  const onNG = useCallback(
    (followedId: string) => {
      if (
        acceptRelationshipRequestMutationData.loading ||
        ngRelationshipRequestMutationData.loading
      ) {
        return;
      }

      ngRelationshipRequestMutation({
        variables: {
          followedID: followedId,
        },
      });
    },
    [
      ngRelationshipRequestMutation,
      acceptRelationshipRequestMutationData.loading,
      ngRelationshipRequestMutationData.loading,
    ]
  );

  return (
    <Plain
      loading={queryResult.loading}
      acceptRequesting={acceptRelationshipRequestMutationData.loading}
      ngRequesting={ngRelationshipRequestMutationData.loading}
      error={queryResult.error}
      onLoadMore={onLoadMore}
      items={items}
      pageInfo={pageInfo}
      onOK={onOK}
      onNG={onNG}
    />
  );
};

export default memo(Connected);
