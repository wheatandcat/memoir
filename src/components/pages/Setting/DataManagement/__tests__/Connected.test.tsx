import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as Recoil from 'recoil';
import * as useFirebaseAuth from 'hooks/useFirebaseAuth';
import * as queries from 'queries/api/index';
import { user } from '__mockData__/user';
import Connected, { Props } from '../Connected';

const propsData = (): Props => ({});

describe('components/pages/DataManagement/Connected.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    jest.spyOn(Recoil, 'useRecoilValue').mockImplementation((): any => ({
      ...user(),
    }));
    jest.spyOn(useFirebaseAuth, 'default').mockImplementation((): any => ({
      setupAuth: jest.fn(),
      onLogout: jest.fn(),
    }));
    jest
      .spyOn(queries, 'useDeleteUserMutation')
      .mockImplementation((): any => [
        jest.fn(),
        { loading: false, error: null },
      ]);
    jest
      .spyOn(queries, 'useRelationshipsQuery')
      .mockImplementation((): any => ({
        loading: false,
        data: {
          relationships: {
            edges: [],
          },
        },
        error: undefined,
        refetch: jest.fn(),
      }));
    wrapper = shallow(<Connected {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
