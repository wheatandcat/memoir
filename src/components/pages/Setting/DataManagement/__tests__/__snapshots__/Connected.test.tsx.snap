// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/DataManagement/Connected.tsx 正常にrenderすること 1`] = `
<Memo(Plain)
  disabledDeleteButton={false}
  error={null}
  loading={false}
  onDelete={[Function]}
/>
`;
